satpy/tests/test_utils.py

```diff
@@ -254,9 +254,28 @@ def test_get_satpos(self, warn_mock):
         warn_mock.assert_called()
 
         # Legacy
+        warn_mock.reset_mock()
         dataset.attrs.pop('orbital_parameters')
         lon, lat, alt = get_satpos(dataset)
         self.assertTupleEqual((lon, lat, alt), (-1, -2, -3))
+        warn_mock.assert_called()
+
+
+@pytest.mark.parametrize(
+    "attrs",
+    (
+        {},
+        {'orbital_parameters':  {'projection_longitude': 1}},
+        {'satellite_altitude': 1}
+    )
+)
+def test_get_satpos_fails_with_informative_error(attrs):
+    """Test that get_satpos raises an informative error message."""
+    dataset = mock.MagicMock(attrs=attrs)
+    with pytest.raises(KeyError) as err:
+        get_satpos(dataset)
+    assert not str(err.value).startswith("satellite_")
+    assert not str(err.value).startswith("orbital_")
```

Comment:
Alternately, you could use `pytest.raises(KeyError, match="...")` here to test for an exact match of the expected error message.

Reply:
I'm moving this to my PR and these don't actually work as intended. The KeyError returned includes a single quote around the missing key so `startswith` never matches it. I'll try using the match regex.